sqlglot/tokens.py

```diff
@@ -1300,8 +1301,12 @@ def _scan_number(self) -> None:
                     self._add(TokenType.NUMBER, number_text)
                     self._add(TokenType.DCOLON, "::")
                     return self._add(token_type, literal)
-                elif self.dialect.IDENTIFIERS_CAN_START_WITH_DIGIT:
-                    return self._add(TokenType.VAR)
+                else:
+                    replaced = literal.replace("_", "")
+                    if self.dialect.NUMBERS_CAN_BE_UNDERSCORE_SEPARATED and replaced.isdigit():
+                        return self._add(TokenType.NUMBER, number_text + replaced)
```

Comment:
What's `number_text + replaced` here? Don't we only need `replaced`?

Reply:
`number_text` would be the first substring, e.g in `123_456_789` we'd have:

- `number_text`: `123`
- `literal`: `_456_789`
- `replaced`: `456789`

Reply:
Gotcha, thanks for clarifying.